src/compiler/program.ts

```diff
@@ -327,6 +327,7 @@ namespace ts {
         let files: SourceFile[] = [];
         let fileProcessingDiagnostics = createDiagnosticCollection();
         const programDiagnostics = createDiagnosticCollection();
+        const emitBlockingDiagnostics = createDiagnosticCollection();

```

Comment:
why do we need this list. is not it sufficient to have the map (`hasEmitBlockingDiagnostics`)?